clients/src/main/java/org/apache/kafka/common/config/provider/FileConfigProvider.java

```diff
@@ -98,7 +95,7 @@ public ConfigData get(String path, Set<String> keys) {
 
     // visible for testing
     protected Reader reader(String path) throws IOException {
-        return new BufferedReader(new InputStreamReader(Files.newInputStream(Paths.get(path)), StandardCharsets.UTF_8));
+        return Files.newBufferedReader(Paths.get(path));
```

Comment:
This should use the variant that takes a Charset, right?  As per https://issues.apache.org/jira/browse/KAFKA-367

Reply:
No, that's not necessary. `Files` does the right thing and uses UTF8 if no encoding is specified. That's why I removed the explicit encoding.